tests/functional/tests.py

```diff
@@ -588,26 +588,72 @@ def test_get_bucket_policy(client, log_output):
         except Exception as err:
             raise Exception(err)
 
-def test_set_bucket_policy(client, log_output):
+def test_set_bucket_policy_readonly(client, log_output):
     # Get a unique bucket_name
     log_output.args['bucket_name'] = bucket_name = generate_bucket_name()
-    log_output.args['prefix'] = prefix = '1/'
+    log_output.args['prefix'] = prefix = ''
     try:
         client.make_bucket(bucket_name)
         # Set read-only policy successfully.
         client.set_bucket_policy(bucket_name, prefix, Policy.READ_ONLY)
+        # Validate if the policy is set correctly
+        policy_name = client.get_bucket_policy(bucket_name)
+        if policy_name != Policy.READ_ONLY:
+            raise ValueError('Failed to set ReadOnly bucket policy: ' + policy_name)
+    except APINotImplemented:
+        print(log_output.json_report(alert='Not Implemented', status=LogOutput.NA))
+    except Exception as err:
+        raise Exception(err)
```

Comment:
shoul;dn't this exception be captured in the log as well?

Reply:
All of the generic exceptions like this one are handled and logged in the main() function.